test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -121,16 +120,13 @@ public void MatrixFactorizationSimpleTrainAndPredict()
 
             // Compute prediction errors
             var metrices = mlContext.Recommendation().Evaluate(prediction, labelColumnName: labelColumnName, scoreColumnName: scoreColumnName);
-
+            var toleranceDelta = Math.Pow(10, -1 * tolerance);
```

Comment:
Would it be better to return the toleranceDelta directly from the GetData method? The user can specify the tolerance as an integer of number of digits. And GetData can return the converted value directly.

Reply:
It appears you need integer values in some places. Can you find a way to standardize the comparison?

Reply:
Hi Harish. Indeed I do need integer values in most places where `VaryingTolerance` is used, which is why I made this `Math.Pow(10, -1 * tolerance)` change in `MatrixFactorizationSimpleTrainAndPredict`. Using `Assert.Equal(double expected, double actual, int precision)` is a nice way to standardize this comparison, however the precision metric here doesn't work well when `double actual` is rounded up or down.

For example, let us have:
`double expected = 0.444445`
`double actual = 0.444444`
`int precision = 6`

Here, `expected` and `actual` are within 1e-6 of accuracy. However, with `Assert.Equal`, `expected` is rounded up to 0.444450, and `actual` is rounded down to 0.444440. As a result, `Assert.Equal` returns `false`, when actually it should return `true`. This is why I went with this approach here. What would you recommend in this case @harishsk ?